core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -68,6 +68,8 @@
 public class MapperService extends AbstractIndexComponent implements Closeable {
 
     public static final String DEFAULT_MAPPING = "_default_";
+    public static final String DYNAMIC_MAPPING_ENABLED_SETTING = "index.mapper.dynamic";
+    public static final boolean DYNAMIC_MAPPING_ENABLED_DEFAULT = true;
```

Comment:
The key used in settings is `index.mapper.dynamic`; I changed the `.` to `_` and made it uppercase.